crates/miden-lib/asm/miden/note.masm

```diff
@@ -83,6 +83,84 @@ export.get_assets
     # => [num_assets, dest_ptr]
 end
 
+#! Returns the number of assets and the assets commitment of the input note by the provided index.
+#!
+#! Assets hash then could be used to get the assets data from the advice map.
+#!
+#! Inputs:  [note_index]
+#! Outputs: [ASSETS_HASH, num_assets]
+#!
+#! Where:
+#! - note_index is the index of the input note which assets info should be returned.
+#! - num_assets is the number of assets in the note which has been requested.
+#! - ASSETS_HASH is the vault hash of the note which has been requested.
+#!
+#! Panics if:
+#! - the note index is greater or equal to the overall number of input notes.
+#!
+#! Invocation: exec
+export.input_note_get_assets_info
```

Comment:
I'm thinking that maybe this procedure (and additional procedures after that), should go into a separate module. Maybe something like `input_notes`? This way we'd be able to access them as:
```
use.miden::input_notes

begin
    exec.input_notes::get_assets_info
end
```

Reply:
For now in order to merge this PR faster I will implement `input_notes` module (and probably `output_notes`), but later I'll create an issue to rework this procedures as you suggested.

Reply:
Mh, maybe I'd just leave this as-is for now to avoid too many big changes in this PR and create an issue to take a look at how to specifically structure this, what do you think?